ui/src/app/components/app-router.spec.tsx

```diff
@@ -73,21 +80,17 @@ const otherAlwaysTrueGuard: Guard = {
 
 const makeAppRouter = () => {
   return <AppRouter>
-    <AppRoute path='/unprotected-route'><TestComponent text={'Unprotected Route'}/></AppRoute>
-    <AppRoute path='/punting'><TestComponent text={'Punting'}/></AppRoute>
-    <ProtectedRoutes guards={[alwaysFalseGuard]}>
-      <AppRoute path='/unreachable-path'><TestComponent text={'Unreachable Path'}/></AppRoute>
-    </ProtectedRoutes>
-    <ProtectedRoutes guards={[alwaysTrueGuard]}>
-      <AppRoute path='/protected-route'><TestComponent text={'Protected Route'}/></AppRoute>
-      <AppRoute path='/other-protected-route'><TestComponent text={'Other Protected Route'}/></AppRoute>
-      <ProtectedRoutes guards={[otherAlwaysTrueGuard]}>
-        <AppRoute path='/nested-protected-route'><TestComponent text={'Nested Protected Route'}/></AppRoute>
-      </ProtectedRoutes>
-      <ProtectedRoutes guards={[alwaysFalseGuard]}>
-        <AppRoute path='/nested-unreachable-path'><TestComponent text={'Unreachable Path'}/></AppRoute>
-      </ProtectedRoutes>
-    </ProtectedRoutes>
+    <Switch>
+      <AppRoute exact path='/unprotected-route'><TestComponent text={'Unprotected Route'}/></AppRoute>
+      <AppRoute exact path='/punting'><TestComponent text={'Punting'}/></AppRoute>
+      <AppRoute exact path='/unreachable-path' guards={[alwaysFalseGuard]}><TestComponent text={'Unreachable Path'}/></AppRoute>
+      <AppRoute exact path='/protected-route' guards={[alwaysTrueGuard]}><TestComponent text={'Protected Route'}/></AppRoute>
+      <AppRoute exact path='/other-protected-route' guards={[alwaysTrueGuard]}><TestComponent text={'Other Protected Route'}/></AppRoute>
+      <AppRoute exact path='/nested-protected-route' guards={[alwaysTrueGuard, otherAlwaysTrueGuard]}><TestComponent text={'Nested Protected Route'}/></AppRoute>
+      <AppRoute exact path='/nested-unreachable-path' guards={[alwaysTrueGuard, alwaysFalseGuard]}><TestComponent text={'Unreachable Path'}/></AppRoute>
+      <AppRoute exact path='/not-found'><TestComponent text={'Not Found'}/></AppRoute>
+      <AppRoute exact={false} path='*'><Redirect to='/not-found'/></AppRoute>
```

Comment:
is the exact=false needed?

Reply:
I removed the `exact=true` default for `AppRoute` because of the component vs. function returning component issue. So now `exact` must be specified one way or the other. Given this works on regex it can probably works with either `exact=true` or `exact=false` but `exact` must be specified.

Reply:
oh ok. I was wondering if it would just default to false if it didn't find one but I think stating it explicitly is more clear anyways. especially, given that we don't use the false version very often.